File: drouter-plugin/src/main/groovy/com/didi/drouter/annotation/Router.java
package com.didi.drouter.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by gaowei on 2019/1/30
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.CLASS)
public @interface Router {

    String scheme() default "";

    String host() default "";

    String path();

    Class[] interceptor() default {};

    int thread() default 0;

    int priority() default 0;

    boolean hold() default false;
}

File: plugins/src/main/kotlin/com/didi/drouter/annotation/Router.kt
package com.didi.drouter.annotation

import kotlin.reflect.KClass

/**
 * Created by gaowei on 2019/1/30
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.BINARY)
annotation class Router(
    val uri: String = "",
    val scheme: String = "",
    val host: String = "",
    val path: String = "",
    val interceptor: Array<KClass<*>> = [],
    val interceptorName: Array<String> = [],
    val thread: Int = 0,
    val priority: Int = 0,
    val hold: Boolean = false
)
